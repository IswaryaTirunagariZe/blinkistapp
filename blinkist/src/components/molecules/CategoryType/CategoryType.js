import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import SvgIcon from '@material-ui/core/SvgIcon';
import { Typography, Box } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    '& > svg': {
      margin: theme.spacing(2),
    },
  },
}));

function HomeIcon(props) {
  return (
    <SvgIcon {...props}>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );
}

export default function CategoryType(props) {
  const classes = useStyles();
  return (
    <Box display="flex" alignItems="center">
      <HomeIcon />
      <h4>
        {props.categoryType}
      </h4>
    </Box>
  );
}