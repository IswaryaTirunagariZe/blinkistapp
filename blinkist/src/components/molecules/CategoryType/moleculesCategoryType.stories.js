import React from 'react';

import CategoryType from './CategoryType';


export default {
  title: 'Molecules/CategoryType',
  component: CategoryType,
};

const CategoryTypeTemp = (args) => <CategoryType {...args} />;


export const CategoryTypeView = CategoryTypeTemp.bind({});
CategoryTypeView.args = {
    categoryType:"Category Type"
};
