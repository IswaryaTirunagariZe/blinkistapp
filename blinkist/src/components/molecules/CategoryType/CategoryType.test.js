import { shallow, render } from "enzyme";

import CategoryType from "./CategoryType";

describe("Categeory List", () => {
  it("Renders Categeory List", () => {
    const categeories = ['Entrepreneurship',
    'Politics',
    'Marketing & Sales',
    'Science',
    'Health & Nutrition',
    'Personal Development',
    'Economics',
    'History',
  ]
    const wrapper = render(<CategoryType categoryType={categeories} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});