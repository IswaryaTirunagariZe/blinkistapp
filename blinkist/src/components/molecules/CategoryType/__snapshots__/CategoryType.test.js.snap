// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categeory List Renders Categeory List 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "MuiBox-root MuiBox-root-2",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "aria-hidden": "true",
          "class": "MuiSvgIcon-root",
          "focusable": "false",
          "viewBox": "0 0 24 24",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "d": "M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z",
            },
            "children": Array [],
            "name": "path",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "d": undefined,
            },
            "x-attribsPrefix": Object {
              "d": undefined,
            },
          },
        ],
        "name": "svg",
        "namespace": "http://www.w3.org/2000/svg",
        "next": Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "data": "EntrepreneurshipPoliticsMarketing & SalesScienceHealth & NutritionPersonal DevelopmentEconomicsHistory",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h4",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-hidden": undefined,
          "class": undefined,
          "focusable": undefined,
          "viewBox": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-hidden": undefined,
          "class": undefined,
          "focusable": undefined,
          "viewBox": undefined,
        },
      },
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "EntrepreneurshipPoliticsMarketing & SalesScienceHealth & NutritionPersonal DevelopmentEconomicsHistory",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h4",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "aria-hidden": "true",
            "class": "MuiSvgIcon-root",
            "focusable": "false",
            "viewBox": "0 0 24 24",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "d": "M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z",
              },
              "children": Array [],
              "name": "path",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "d": undefined,
              },
              "x-attribsPrefix": Object {
                "d": undefined,
              },
            },
          ],
          "name": "svg",
          "namespace": "http://www.w3.org/2000/svg",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-hidden": undefined,
            "class": undefined,
            "focusable": undefined,
            "viewBox": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-hidden": undefined,
            "class": undefined,
            "focusable": undefined,
            "viewBox": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
